web3.js/src/connection.ts

```diff
@@ -1524,6 +1524,54 @@ type RootSubscriptionInfo = {
   subscriptionId: SubscriptionId | null; // null when there's no current server subscription id
 };
 
+/**
+ * @internal
+ */
+const LogsResult = pick({
+  err: TransactionErrorResult,
+  logs: array(string()),
+  signature: string(),
+});
+
+/**
+ * Logs result.
+ *
+ * @typedef {Object} Logs.
+ */
+export type Logs = {
+  err: TransactionError | null;
+  logs: string[];
+  signature: string;
+};
+
+/**
+ * Expected JSON RPC response for the "logsNotification" message.
+ */
+const LogsNotificationResult = pick({
+  result: notificationResultAndContext(LogsResult),
+  subscription: number(),
+});
+
+/**
+ * Filter for log subscriptions.
+ */
+export type LogsFilter = {mentions: string[]} | 'all ' | 'allWithVotes';
```

Comment:
Currently the logs pubsub api [only supports one mention](https://github.com/solana-labs/solana/blob/2c4ff000b9dc28b5e9dc89c016e67aa1f3440825/core/src/rpc_pubsub.rs#L344), so I think making that explicit for now would be better (we can expand on the type union later to allow an array). Also, can you change the mention type from string to `PublicKey` and then handle the base58 conversion internally?